### DISCUSSION TITLE
Prisma in docker connecting to localhost MySQL database

### DISCUSSION THREAD

**Discussion by FuelKubitox** [prisma/prisma#6659] [Q&A] [answered]
At the moment im trying to use the bridge network mode with docker for my Containers. Im using host mode and everything works fine, but i want bridge mode for more abstraction.

I created a network in docker and with docker-compose i join these network, when i create the image. Creating and starting the container works fine, but i cant connect to my MySQL Database which is running on the host.

When i go in to the shell from the docker, which is running prisma, then i cant connect to the database. I tried to connect from the docker to the gateway to port 3306 and i get the error that prisma cant reach the database.

My connection string looks like this

`DATABASE_URL="mysql://USER:PASSWORD@HOSTNAME:3306/database`

The hostname get passed in docker-compose as extra_hosts and is the IP from the network gateway i created. I also tried to connect to the host IP. But cant reach the database.

Then i tried to map a volume to the container where the mysqld.sock is and tried to connect to the database, but i get the same error. Cant reach the database. The string for that looks like this.

`DATABASE_URL="mysql://USER:PASSWORD@localhost:3306/database?socket=/app/mysql`

In the docker the sock file is in /app/mysql, but i cant connect. I tried many things, but at the end im thinking that this is maybe a prisma special thing or problem.

Someone can explain me how i can archive it what im trying to do?

**Comment by pablopsp** [comment]
Have you tried with the name of your database docker container as the hostname of the connection string? 


```docker
# docker-compose.yml
services:
  db:
```

Then `DATABASE_URL="mysql://USER:PASSWORD@db:3306/database`

**Reply by FuelKubitox** [reply]
Why? The database isnt in the docker container. The database runs on the host of my machine.

The weird thing is, that i hoped that the prisma client can connect to the mysqld.sock, but that doesnt work either. I the mysqld.sock file is mapped in the docker container over volumes.

When i use the hostname of the docker, then the prisma client try to connect to localhost in the container or am i wrong?

**Comment by 2color** [comment, marked as answer]
Hey @FuelKubitox,

If I understand you correctly, you are trying to access the MySQL database on your host machine from the Docker container. Doing this with Docker has changed a bit over the years. Check out this [answer](https://stackoverflow.com/a/43541732/1430687) on StackOverflow. 

Depending on the OS you're using, you should be able to access the host machine with the following DNS host: `host.docker.internal` which Docker adds to the container pointing to the host machine's IP.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
